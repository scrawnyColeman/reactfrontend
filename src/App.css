html {
    background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url(./images/bgimg.jpg) no-repeat fixed;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
}
@import url('https://fonts.googleapis.com/css?family=Didact+Gothic');

body {
    color: #eee;
    font-family: 'Century Gothic', 'Futura', 'Didact Gothic', san-serif;
}

.App {
    text-align: center;
}
