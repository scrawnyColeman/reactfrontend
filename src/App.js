import React from 'react';
import styled from 'styled-components';
import './App.css';
import Login from './components/Authentication/Login.jsx';
import Footer from './components/Footer/component';
import Header from './components/Header/component';
import ForumPage from './components/ForumPage/component';
import UserPage from './components/UserPage/component';
import { footerHeight, headerHeight } from './constants/measures';

const Page = styled.div`
    display: grid;
    grid-template-rows: ${headerHeight} 1fr ${footerHeight};
`;

const App = () => (
    <Page className="App">
        {window.location.pathname === '/login' && <Login />}
        <Page>
            <Header userName={'Thomas'} />
            {window.location.pathname === '/home' && <UserPage />}
            {window.location.pathname === '/forum' && <ForumPage />}
            <Footer />
        </Page>
    </Page>
);

export default App;
