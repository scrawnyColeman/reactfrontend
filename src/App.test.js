import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import App from './app';

Enzyme.configure({ adapter: new Adapter() });

describe('App component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot();
    });
});
