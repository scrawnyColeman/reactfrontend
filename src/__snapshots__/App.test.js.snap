// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component component matches snapshot 1`] = `
<BrowserRouter>
  <styled.div
    className="App"
  >
    <Header
      userName="ScrawnyColeman"
    />
    <BrowserRouter
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/home/:name"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/home"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/learn"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/forum"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/forum/:id"
      />
    </BrowserRouter>
    <Footer />
  </styled.div>
</BrowserRouter>
`;
